### PR TITLE
Issue 213 notification system

### LINKED ISSUE #213: Add User Post Subscriptions and Notifications
Create Board Notifications tool for receiving notifications, for instance, upon:
- Comment, upvote, or tag by another user to their posts

Board Notifications could include a Bell icon in the top navigation bar with a numerical counter indicating the number of notifications. Once clicked, a tray could be displayed that shows a list of notifications. Selecting a notification could open the modal pop-up for that post. 

SCORE has something like this:
Screen Shot 2022-03-22 at 9 07 22 PM

### PR DESCRIPTION
**Pull request by l3n0ire**
## Details

- User will be notified whenever a user comments, tags or likes their posts
  - no option to turn on/off notifications yet (should I add this?)
- When a user clicks on a notification, the post modal opens and the notification is marked as viewed
- For now notifications are deleted after they have been viewed
- Mark all as read will mark all notifications as read

Closes #213

### CODE REVIEW

**Comment by JoelWiebe** [comment]
@l3n0ire I like the way you've designed it. We should be good with everyone getting system notifications, maybe later if we have email notifications, those can be toggled off in some user settings. Deleting notifications after they are read is good for now and will help keep the notifications pane clean. Great work.

**Comment by JoelWiebe** [comment]
@l3n0ire Am I looking in the right place for notifications? Should notifications be displayed in the top nav bar?

Screen Shot 2022-04-12 at 6 31 10 PM

**Comment by l3n0ire** [comment]
@JoelWiebe I added the post names and fixed the bugs you mentioned. If the postname is really long and cannot fit on one line by itself, then it will be shortened, otherwise words that can't fit on a line will just be pushed to the next line

File: src/app/components/canvas/canvas.component.html [hunk 1]
```diff
@@ -19,6 +19,7 @@
         matTooltip="Adjust Settings">
         <mat-icon>settings</mat-icon>
       </button>
+      <app-notification-dropdown [props]="{user:user, board:board}"></app-notification-dropdown>
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.html:22]: Small change but can we do something like `[user]="user" [board]="board"` instead?

File: src/app/components/notification-dropdown/notification-dropdown.component.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  initGroupEventsListener() {
+    const unsubPosts = this.notificationService.observable(
+      this.props.user.id, this.handleNotificationUpdate
+    );
+    return [unsubPosts];
+  }
+
+  handleNotificationUpdate = (notification:Notification) =>{
+    console.log("ran")
```
Comment (Ammar-T) [inline src/app/components/notification-dropdown/notification-dropdown.component.ts:47]:
```suggestion
```

File: src/app/services/canvas.service.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        let post = data.docs[0].data();
+        let notification:Notification = notificationFactory(post.userID,post.postID);
+        let user = await this.userService.getOneById(like.likerID)
+        notification.text = user?.username +" liked \""+ post.title+"\"";
+        await this.notificationService.add(notification);
+
+      
+    }
+
+    async modifyTag(userId:string,postId: string, value: object,): Promise<any>{
```
Comment (Ammar-T) [inline src/app/services/canvas.service.ts:54]:
```suggestion
    async modifyTag(postId: string, value: object,): Promise<any>{
```
We can remove this and get the current user from `auth.service.ts`.

**Comment by Ammar-T** [comment]
Aside from those changes, good work!

**Review by Ammar-T — APPROVED** [review]

**Pull request merged**